Modules/_io/textio.c

```diff
@@ -1301,6 +1301,13 @@ textiowrapper_write(textio *self, PyObject *args)
     Py_DECREF(text);
     if (b == NULL)
         return NULL;
+    if (!PyBytes_Check(b)) {
```

Comment:
Unicode should be accepted too. `_PyBytes_Join()` in 2.7 supports unicode objects.